community/kernel/src/main/java/org/neo4j/kernel/impl/store/LongerShortString.java

```diff
@@ -801,9 +801,11 @@ public static String decode( PropertyBlock block )
 
     private static LongerShortString getEncodingTable( int encodingHeader )
     {
-        final LongerShortString encoding = ENCODINGS_BY_ENCODING[encodingHeader];
-        if (encoding==null) throw new IllegalArgumentException( "Invalid encoding '" + encoding + "'" );
-        return encoding;
+        if ( encodingHeader < 0 | ENCODINGS_BY_ENCODING.length <= encodingHeader )
+        {
+            return null;
```

Comment:
That will never happen because the decoding happens after the `PropertyBlock` data has been consistently read from the page cache. I'll add a descriptive `assert` about this, though.